service/history/shard/context.go

```diff
@@ -116,7 +117,7 @@ type (
 		GetWorkflowExecution(ctx context.Context, request *persistence.GetWorkflowExecutionRequest) (*persistence.GetWorkflowExecutionResponse, error)
 		// DeleteWorkflowExecution add task to delete visibility, current workflow execution, and deletes workflow execution.
 		// If branchToken != nil, then delete history also, otherwise leave history.
-		DeleteWorkflowExecution(ctx context.Context, workflowKey definition.WorkflowKey, branchToken []byte, closeExecutionVisibilityTaskID int64, stage *tasks.DeleteWorkflowExecutionStage) error
```

Comment:
I am only modifying DeleteWorkflowExecution to take lock priority as an argument. We just need that for deprioritizing retention timers. If needed, I can add that to other functions as well.

Reply:
in general we are trying to use low priority for requests for hot workflows? Just want to see if we are on the same page for the overall goal here. Starting with retention timer makes sense.

Reply:
We want to use low priority for things like retention timers, migration, etc. so that it will block live customer traffic.
Currently, this has two priority levels. We can add more levels if we need to. But for this purpose, I think two is enough.